registries/adapters/oauth/client.go

```diff
@@ -81,7 +81,16 @@ func (c *Client) NewRequest(path string) (*http.Request, error) {
 	if err != nil {
 		return nil, err
 	}
-	req.URL.Path = path
+
+	// path might be a fully qualified URL and we only want the path
+	// and query bits
+	thebits, err := url.Parse(path)
```

Comment:
personally, I would change the variable name but not a blocker

Reply:
changed to `pathAsUrl` let me know if that is not okay.

Reply:
actually changed it to `pathAsURL` since `pathAsUrl` pissed off lint :)